src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,109 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int INITIAL_CAPACITY = 16;
+    private static final float LOAD_FACTOR = 0.75f;
+    private static final int MULTIPLIER_CAPACITY = 2;
+    private int size;
+    private Node<K, V>[] table;
+
+    public MyHashMap() {
+        table = (Node<K, V>[]) new Node[INITIAL_CAPACITY];
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size == threshold()) {
+            resize();
+        }
+        Node<K, V> identicalNode = findNode(key);
+        if (identicalNode != null) {
+            identicalNode.value = value;
+        } else {
+            insertNewValue(key, value);
+        }
     }
 
     @Override
     public V getValue(K key) {
-        return null;
+        Node<K, V> node = findNode(key);
+        return node == null ? null : node.value;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private int hash(K key) {
+        return (key == null) ? 0 : Math.abs(key.hashCode()) % table.length;
+    }
+
+    private int threshold() {
+        return (int) (table.length * LOAD_FACTOR);
+    }
+
+    private void resize() {
+        int newCapacity = table.length * MULTIPLIER_CAPACITY;
+        Node<K, V>[] newTable = (Node<K, V>[]) new Node[newCapacity];
```

Comment:
Diamond operator: Use the diamond operator carefully. It's recommended to use `new Node<>[newCapacity]` to avoid unchecked cast warnings.